cmd/transcriber/call/live_captions.go

```diff
@@ -0,0 +1,323 @@
+package call
+
+import (
+	"fmt"
+	"github.com/mattermost/calls-transcriber/cmd/transcriber/apis/whisper.cpp"
+	"github.com/mattermost/calls-transcriber/cmd/transcriber/config"
+	"github.com/mattermost/calls-transcriber/cmd/transcriber/transcribe"
+	"github.com/streamer45/silero-vad-go/speech"
+	"log/slog"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	// TODO: these need to be in cfg and env var settable, and in proper style.
+	parallelTranscribersPool = 4
+	threadsPerTranscriber    = 1
+
+	chunkSizeInMs            = 1000
+	maxWindowSizeInMs        = 10000
+	audioDataChannelBuffer   = 30 // so we don't block while processing the window's vad
+	removeWindowAfterSilence = 3 * time.Second
+
+	// VAD settings
+	vadWindowSizeInSamples  = 512
+	vadThreshold            = 0.5
+	vadMinSilenceDurationMs = 300
+	vadMinSpeechDurationMs  = 200
+	vadSilencePadMs         = 32
+)
+
+// TODO: belongs in calls-common?
```

Comment:
This probably belongs in either the `rtcd` client or https://github.com/mattermost/mattermost-plugin-calls/tree/main/server/public . At some point I'd like to unify the two.

Reply:
Nope, you were right, I put it in public: https://github.com/mattermost/mattermost-plugin-calls/pull/633/commits/9055277fda92c143acfd66eee72d2a63021c327e